_posts/2016-07-07-better-email-validation-with-obey-and-mailgun.markdown

```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+title: "Better Email Validation with Obey and Mailgun"
+date: 2016-07-07 10:00:00
+categories: asynchronous validation nodejs
+author: kent.safranski
+cover: /assets/images/covers/books.jpg
+description: Email validation seems simple, but you really should be using a service to accurately validate addresses. Here's how Obey and Mailgun can work together to reliably validate emails.
+comments: true
+---
+
+Validating email is simple, right? Just throw a regex at it? Well it's not as simple as you might think. The RFC Spec (Page 27 of [this monster](https://www.ietf.org/rfc/rfc0822.txt), if you're interested) is clear as mud on what constitutes a valid email.
+
+Basically, with all the special characters and conditions of what can go before and after the `@`, a regex to validate emails accurately would be incomprehensable at best. 
+
+_If you'd like to find out more about why email validation sucks check out [this article](http://www.kbedell.com/2011/03/16/how-to-validate-an-email-address-using-regular-expressions/), for now let's move along._
+
+### Using a Service for Validation
+
+We want to know if an email is valid, but a regex won't work, so our best option is a service. [Mailgun's API for email validation](https://documentation.mailgun.com/api-email-validation.html#email-validation) is a wonderful option. It will check a number of things for us:
```

Comment:
Might be worth it to explore the possibility of a library before jumping to a service, as I would imagine most people would read this and go "What? Why make an API call when there are 1000 libraries for this?". Might lend a lot more credibility to pull out some popular libraries and talk about how well-liked and great they are, then pull a "but they have a common shortcoming..."